### PR TITLE
Fix startPolling error handling

### PR DESCRIPTION
**Pull request by ritch**
/to @rmg 

This allows for arc to provide a callback but its not required.

### CODE REVIEW

File: proxy/boot/manager-host.js [hunk 1]
```diff
@@ -30,11 +30,21 @@ module.exports = function setupHooks(server) {
...
-    setInterval(function() {
-      ManagerHost.sync();
-    }, ManagerHost.settings.interval || 1000);
+  ManagerHost.startPolling = function(cb) {
+    var timer;
+    ManagerHost.sync(function(firstErr) {
+      if(firstErr) {
+        return cb ? cb(err) : null;
+      }
+      timer = setInterval(function() {
```
Comment (rmg) [inline proxy/boot/manager-host.js]: Should this timer be `unref()`'d so that it doesn't keep an otherwise "finished" app from exiting?

Comment (ritch) [inline proxy/boot/manager-host.js]: You mean even in the non error case?

Comment (rmg) [inline proxy/boot/manager-host.js]: I mean in the case where the app that is using strong-mesh-client is finished. There's no stopPolling, so the event loop will remain alive unless someone kills the process or the app calls `process.exit()`. This kind of problem comes up all the time with CLI apps and in tests.

File: proxy/server.js [hunk 2]
```diff
@@ -51,7 +51,14 @@ module.exports = function createServer(configFile, options) {
 
   server.use(loopback.rest());
 
-  ManagerHost.startPolling();
+  cb = cb || function defaultPollingCallback(err) {
+    if(err) {
+      err.message = 'Manager Host Polling ' + err.message;
+      throw err;
+    }
+  }
```
Comment (rmg) [inline proxy/server.js]: I believe the coding style in this repo requires semicolons after expressions.

**Comment by rmg** [comment]
Minor nit about a missing semicolon, but otherwise LGTM.

The lack of `Timer.unref()` isn't new, so no need to fix it in this PR.

File: proxy/boot/manager-host.js (hunk 1 above)
Comment (ritch) [inline proxy/boot/manager-host.js]: Sure but I think that is outside the scope of this PR.

**Comment by ritch** [comment]
test please
